x-pack/modules/azure/configuration/logstash/azure.conf.erb

```diff
@@ -0,0 +1,58 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+input{
+<%=
+require 'azure_module_config_generator'
+config_generator = LogStash::Azure::ConfigGenerator.new
+config_generator.generate_input(@settings)
+%>
+}
+
+filter {
+  json {
+    source => "message"
+  }
+
+  if "_jsonparsefailure" not in [tags] {
+    split {
+      field => "records"
+      target => "data"
+    }
+
+    if "_split_type_failure" not in [tags] {
+      ruby{
+        code => "
+          event.get('data').each { |k,v|
+            event.set(k, v)
+          }
+          event.remove('data')
```

Comment:
I can do this and agree with doing as little in a ruby in filter ... but in this case, it is functionally part of what the ruby filter is doing (moving around the data field) . I think this belongs here. If I were to write this as a method it would be contained in a single method. Breaking it apart feels like we splitting the logic of what this does into two locations.